backend/graphql/usecases/create-order.use-case.ts

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { OrderModel } from '../../database/models/order.model'
+import { ProductModel } from '../../database/models/product.model'
+import { NotFoundError } from '../errors/notFoundError'
+import { type CreateOrderInput } from '../inputs/order.input'
+import { type Context } from '../types/context.type'
+import { type Order } from '../types/order.type'
+
+export class CreateOrderUseCase {
+  async execute(input: CreateOrderInput, context: Context): Promise<Order> {
+    const user = context.user
+
+    if (!user) {
+      throw new NotFoundError('User not found')
+    }
+
```

Comment:
- you don't need to replicate Mongo's pattern of `_id`
- this is actually a list of ids so it should be named as `productsIds` or somthing like that